chapters/03-talking_to_servers/readme.md

```diff
@@ -0,0 +1,678 @@
+manually, Rails does that for us.
+
+```html
+<h1>Solve a polynomial!</h1>
+
+<%= form_tag "/polynomials", method: :post do %>
+  <p><b>a</b> * X<sup>2</sup> + <b>b</b> * X + <b>c</b> = <b>d</b></p>
+
+  <p>
+    <%= label_tag(:a, 'a:') %>
+    <%= number_field :a %>
+  </p>
+  <p>
+    <%= label_tag(:b, 'b:') %>
+    <%= number_field :b %>
+  </p>
+  <p>
+    <%= label_tag(:c, 'c:') %>
+    <%= number_field :c %>
+  </p>
+  <p>
+    <%= label_tag(:d, 'd:') %>
+    <%= number_field :d %>
+  </p>
+
+  <button type="submit">Solve!</button>
+<% end %>
+```
+
+Now we can combine these helpers with our knowledge from previous lectures and
+use a loop! That way we don't have to repeat our selfs four times to create this
+form.
+
+```html
+<h1>Solve a polynomial!</h1>
+
+<%= form_tag "/polynomials", method: :post do %>
+  <p><b>a</b> * X<sup>2</sup> + <b>b</b> * X + <b>c</b> = <b>d</b></p>
+
+  <% [:a, :b, :c, :d].each do |param| %>
+    <p>
+      <%= label_tag(param, "#{param}:") %>
+      <%= number_field nil, param %>
+    </p>
+  <% end %>
+
+  <button type="submit">Solve!</button>
+<% end %>
+```
+
+This is now much cleaner! and it still works!
+But we have ways of creating forms in an even easier manner!
+
+Now, let's add the ability to add comments to our posts! We could do use
+Rails' form tags and  helpers to accomplish this:
+
+```html
+<h1>Solve a polynomial!</h1>
+
+<%= form_tag "/comments", method: :post do %>
+  <%= hidden_tag(param, "#{param}:") %>
+
+  <p>
+    <%= label_tag(:content, "Comment:") %>
+    <%= text_area_tag(:content) %>
+  </p>
+
+  <button type="submit">Solve!</button>
+<% end %>
+```
+
+But there is a better way. When working with models, something that will be
+covered in the next chapter, we can use additional helpers which respect the
+context of the model. Meaning, if we want a `text_area` for the Comment's
+`content` attribute we could just ask the form for a `text_area` and it would
+infer that it's for our Comment.
+
+To tap into this we need to change our form a little bit, instead of a
+`form_tag` we are going to use `form_for`. Instead of an URL it accepts an
+instance of a model and outputs a form builder,
+like so `<%= form_for Comment.new do |form_builder| %>`. Now our form is
+aware of the `Comment.new` context and all tags created with it will
+automatically be in the context of and bear attribute names of the `Comment`.
+E.g.:
+
+```html
+<%= form_for Comment.new do |f| %>
+  <%= f.text_area :content, size: '60x10' %>
+  <%= f.submit 'Post' %>
+<% end %>
+```
+
+If we now add this snippet at the bottom of the Post's show page, we can create
+comments from that page.
+
+```html
+<p id="notice"><%= notice %></p>
+
+<p>
+  <strong>Author:</strong>
+  <%= @post.author_id %>
+</p>
+
+<p>
+  <strong>Content:</strong>
+  <%= @post.content %>
+</p>
+
+<p>
+  <strong>Published:</strong>
+  <%= @post.published %>
+</p>
+
+<p>
+  <strong>Comments:</strong>
+</p>
+
+<p>
+  <br />
+  <strong>Post a comment:</strong>
```

Comment:
Maybe link post id and add field to put in author manually.

Reply:
Then we have to explain relationships else it doesn't make sense

Reply:
We can explain this along with belongs_to and has_many

Reply:
:+1: